util/noeintr_close_testing.h

```diff
@@ -0,0 +1,22 @@
+#ifndef NOEINTR_CLOSE_TESTING_H_
```

Comment:
I'm not certain if you'll like having `noeintr_close_testing.{c,h}` as separate files.  I found it useful to shove it away from `noeintr_close.c` itself, but I suspect that you might prefer it in the same file.

One downside of combining them: we'd need to put `void noeintr_close_testing_print_stats(void);` in `noeintr.h` (protected by `#ifdef NOEINTR_CLOSE_TESTING`), and that looks a bit weird.  Or we could go with what I did temporarily in my devel code: dump the `void noeintr_close_testing_print_stats(void);` declaration in `tests/noeintr_close/*/main.c` along with a comment "keep this declaration in sync with the actual code in `noeintr_close.c`".  The latter idea feels really icky, but at least it would avoid polluting `noeintr.h` with a test-only function.